pandas/tests/frame/test_timeseries.py

```diff
@@ -63,3 +63,22 @@ def test_datetime_assignment_with_NaT_and_diff_time_units(self):
             {0: [1, None], "new": [1e9, None]}, dtype="datetime64[ns]"
         )
         tm.assert_frame_equal(result, expected)
+
+    def test_slice_irregular_datetime_index_with_nan(self):
```

Comment:
move these to pandas/tests/indexing/test_partial.py

*always* use
result=
expected=
tm.assert_.....